### ISSUE TITLE
Problem installing with perl 5.30.1

### ISSUE THREAD

**Issue by rabbiveesh** [bucardo/dbdpg#67]
I'm trying to install DBD::Pg with perl 5.30.1 under perlbrew (I have it installed normally on my 5.22 system perl). I'm getting the following build log:

```
cpanm (App::cpanminus) 1.7044 on perl 5.030001 built for x86_64-linux
Work directory is /home/veesh/.cpanm/work/1584234914.30106
You have make /usr/bin/make
You have /usr/bin/wget
You have /bin/tar: tar (GNU tar) 1.28
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching DBD::Pg () on cpanmetadb ...
--> Working on DBD::Pg
Fetching http://www.cpan.org/authors/id/T/TU/TURNSTEP/DBD-Pg-3.10.4.tar.gz
-> OK
Unpacking DBD-Pg-3.10.4.tar.gz
Entering DBD-Pg-3.10.4
Checking configure dependencies from META.yml
Checking if you have DBI 1.614 ... Yes (1.643)
Checking if you have version 0 ... Yes (0.9924)
Configuring DBD-Pg-3.10.4
Running Makefile.PL
Configuring DBD::Pg 3.10.4
PostgreSQL version: 100011 (default port: 5432)
POSTGRES_HOME: (not set)
POSTGRES_INCLUDE: /usr/include/postgresql
POSTGRES_LIB: /usr/lib/x86_64-linux-gnu
OS: linux
Checking if your kit is complete...
Looks good
Using DBI 1.643 (for perl 5.030001 on x86_64-linux) installed in /home/veesh/perl5/lib/perl5/x86_64-linux/auto/DBI/
Generating a Unix-style Makefile
Writing Makefile for DBD::Pg
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Test::More 0.88 ... Yes (1.302171)
Checking if you have version 0 ... Yes (0.9924)
Checking if you have ExtUtils::MakeMaker 6.11 ... Yes (7.30)
Checking if you have DBI 1.614 ... Yes (1.643)
Checking if you have Time::HiRes 0 ... Yes (1.9760)
Building and testing DBD-Pg-3.10.4
cp Pg.pm blib/lib/DBD/Pg.pm
cp lib/Bundle/DBD/Pg.pm blib/lib/Bundle/DBD/Pg.pm
Running Mkbootstrap for Pg ()
chmod 644 "Pg.bs"
"/home/veesh/perl5/perlbrew/perls/perl-5.30.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Pg.bs blib/arch/auto/DBD/Pg/Pg.bs 644
"/home/veesh/perl5/perlbrew/perls/perl-5.30.1/bin/perl" -p -e "s/~DRIVER~/Pg/g; s/^do\(/dontdo\(/" /home/veesh/perl5/lib/perl5/x86_64-linux/auto/DBI/Driver.xst > Pg.xsi
"/home/veesh/perl5/perlbrew/perls/perl-5.30.1/bin/perl" "/home/veesh/perl5/lib/perl5/ExtUtils/xsubpp"  -typemap '/home/veesh/perl5/perlbrew/perls/perl-5.30.1/lib/5.30.1/ExtUtils/typemap'  Pg.xs > Pg.xsc
mv Pg.xsc Pg.c
cc -c  -I/usr/include/postgresql -I/home/veesh/perl5/lib/perl5/x86_64-linux/auto/DBI -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DPGLIBVERSION=100011 -DPGDEFPORT=5432 -O2  -DPERL_EXTMALLOC_DEF -Dmalloc=Perl_malloc -Dfree=Perl_mfree -Drealloc=Perl_realloc -Dcalloc=Perl_calloc -DVERSION=\"3.10.4\" -DXS_VERSION=\"3.10.4\" -fPIC "-I/home/veesh/perl5/perlbrew/perls/perl-5.30.1/lib/5.30.1/x86_64-linux/CORE"   Pg.c
cc -c  -I/usr/include/postgresql -I/home/veesh/perl5/lib/perl5/x86_64-linux/auto/DBI -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DPGLIBVERSION=100011 -DPGDEFPORT=5432 -O2  -DPERL_EXTMALLOC_DEF -Dmalloc=Perl_malloc -Dfree=Perl_mfree -Drealloc=Perl_realloc -Dcalloc=Perl_calloc -DVERSION=\"3.10.4\" -DXS_VERSION=\"3.10.4\" -fPIC "-I/home/veesh/perl5/perlbrew/perls/perl-5.30.1/lib/5.30.1/x86_64-linux/CORE"   dbdimp.c
cc -c  -I/usr/include/postgresql -I/home/veesh/perl5/lib/perl5/x86_64-linux/auto/DBI -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DPGLIBVERSION=100011 -DPGDEFPORT=5432 -O2  -DPERL_EXTMALLOC_DEF -Dmalloc=Perl_malloc -Dfree=Perl_mfree -Drealloc=Perl_realloc -Dcalloc=Perl_calloc -DVERSION=\"3.10.4\" -DXS_VERSION=\"3.10.4\" -fPIC "-I/home/veesh/perl5/perlbrew/perls/perl-5.30.1/lib/5.30.1/x86_64-linux/CORE"   quote.c
cc -c  -I/usr/include/postgresql -I/home/veesh/perl5/lib/perl5/x86_64-linux/auto/DBI -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DPGLIBVERSION=100011 -DPGDEFPORT=5432 -O2  -DPERL_EXTMALLOC_DEF -Dmalloc=Perl_malloc -Dfree=Perl_mfree -Drealloc=Perl_realloc -Dcalloc=Perl_calloc -DVERSION=\"3.10.4\" -DXS_VERSION=\"3.10.4\" -fPIC "-I/home/veesh/perl5/perlbrew/perls/perl-5.30.1/lib/5.30.1/x86_64-linux/CORE"   types.c
rm -f blib/arch/auto/DBD/Pg/Pg.so
LD_RUN_PATH="/usr/lib/x86_64-linux-gnu" cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong  Pg.o dbdimp.o quote.o types.o  -o blib/arch/auto/DBD/Pg/Pg.so  \
   -L/usr/lib/x86_64-linux-gnu -lpq -lm   \
  
chmod 755 blib/arch/auto/DBD/Pg/Pg.so
Manifying 2 pod documents
"/home/veesh/perl5/perlbrew/perls/perl-5.30.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Pg.bs blib/arch/auto/DBD/Pg/Pg.bs 644
PGINITDB="/usr/lib/postgresql/10/bin/initdb" PERL_DL_NONLAZY=1 "/home/veesh/perl5/perlbrew/perls/perl-5.30.1/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00_signature.t ....... skipped: Set the environment variable TEST_SIGNATURE to enable this test
t/00basic.t ............ ok
# 
# DBI                         Version 1.643
# DBD::Pg                     Version 3.10.4
# Perl                        Version 5.30.1
# OS                          linux
# PostgreSQL (compiled)       100011
# PostgreSQL (target)         100011
# PostgreSQL (reported)       PostgreSQL 10.11 (Ubuntu 10.11-1.pgdg16.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609, 64-bit
# Default port                5432
# DBI_DSN                     dbi:Pg:
# DBI_USER                    postgres
# Test schema                 dbd_pg_testschema
# LANG                        en_US.UTF-8
# array_nulls                 on
# backslash_quote             safe_encoding
# client_encoding             UTF8
# server_encoding             UTF8
# standard_conforming_strings on
# Adjusted:                   DBI_DSN 
t/01connect.t .......... ok
t/01constants.t ........ ok
t/02attribs.t .......... ok

#   Failed test 'DB handle method "table_info" returns correct number of rows when given a 'MATERIALIZED VIEW' type argument'
#   at t/03dbmethod.t line 659.
#          got: '2'
#     expected: '0'

#   Failed test 'DB handle method "table_info" returns correct number of rows when given a 'MATERIALIZED VIEW' type argument'
#   at t/03dbmethod.t line 674.
#          got: '3'
#     expected: '1'
# Looks like you failed 2 tests of 580.
t/03dbmethod.t ......... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/580 subtests 
	(less 5 skipped subtests: 573 okay)
t/03smethod.t .......... ok
t/04misc.t ............. ok
t/06bytea.t ............ ok
t/07copy.t ............. ok
t/08async.t ............ ok
t/09arrays.t ........... ok
t/10_pg_error_field.t .. ok
t/12placeholders.t ..... ok
t/20savepoints.t ....... ok
t/30unicode.t .......... ok
t/99cleanup.t .......... ok

Test Summary Report
-------------------
t/03dbmethod.t       (Wstat: 512 Tests: 580 Failed: 2)
  Failed tests:  123, 125
  Non-zero exit status: 2
Files=17, Tests=2860, 26 wallclock secs ( 0.48 usr  0.03 sys +  4.58 cusr  0.56 csys =  5.65 CPU)
Result: FAIL
Failed 1/17 test programs. 2/2860 subtests failed.
Makefile:985: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
-> FAIL Installing DBD::Pg failed. See /home/veesh/.cpanm/work/1584234914.30106/build.log for details. Retry with --force to force install it.

```

**Comment by rabbiveesh** [comment]
I'm suspecting that this has to do with my local pg server having 2 materialized views in it...
Yeah, and after testing it's clearly fixed by having table_info only search within the test schema that we're using. I'm gonna post a patch in a few minutes.

**Comment by rabbiveesh** [comment]
Eh, nevermind, I'm not actually sure how to patch it, I'd just like to point out that it's accidentally checking all schemas, not just the test schema.

**Comment by turnstep** [comment]
Thanks for the report. Was able to duplicate the issue and it should be fixed in c414c8cf2cdafff8a5c892f0143cf433e3b89bd9. Would appreciate if you could confirm it works for you.

**Comment by rabbiveesh** [comment]
Confirmed that it's fixed! Many thanks
